artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java

```diff
@@ -974,6 +974,7 @@ public void run() {
             } catch (Throwable e) {
                result.fail(e);
                logger.error("appendDeleteRecord:" + e, e);
+               setErrorCondition(callback, null, e);
```

Comment:
I understand you fixed a synchronization here, but I'm a bit concerned into forcing a failure on the callback when non IO Errors have happened.

Do you really need this line?

Reply:
It is handled like this on the other append calls, let me find one...

Reply:
https://github.com/apache/activemq-artemis/blob/47e17ee754044ec1b2bcf0f1712f71a994ed0009/artemis-journal/src/main/java/org/apache/activemq/artemis/core/journal/impl/JournalImpl.java#L860